src/app/content/components/Page/PageComponent.tsx

```diff
@@ -77,40 +87,60 @@ export default class PageComponent extends Component<PagePropTypes, PageState> {
 
     const shouldUpdateHighlights = prevProps !== this.props ||
       (prevState.hasSearchError === this.state.hasSearchError &&
-        prevState.selectedSearchResultId === this.state.selectedSearchResultId);
+        prevState.highlightId === this.state.highlightId);
 
     if (!shouldUpdateHighlights) { return; }
 
-    const highlgihtsAddedOrRemoved = this.highlightManager.update();
+    const highlgihtsAddedOrRemoved = this.highlightManager.update({ onSelect: this.onHighlightSelect });
 
     this.searchHighlightManager.update(prevProps.searchHighlights, this.props.searchHighlights, {
       forceRedraw: highlgihtsAddedOrRemoved,
-      onSelect: this.onHighlightSelect,
+      onSelect: this.onSearchHighlightSelect,
     });
   }
 
-  public onHighlightSelect: OptionsCallback = ({current, selectedHighlight}) => {
+  public onHighlightSelect = (scrollTarget: HighlightScrollTarget | null, highlight: Highlight | null) => {
```

Comment:
i don't feel like we should need to have `onHighlightSelect` or `onSearchHighlightSelect` in PageComponent. can't all that logic be in the managers? 

PageComponent is handling the errors, but i don't think it has to have case by case logic or awareness, you could pass in functions like `clearError`, `setError(subjectId: string, messageKey: string)` into the managers for them to be able to communicate that back.